client/components/article_list/index.jsx

```diff
@@ -0,0 +1,127 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import moment from 'moment'
+import { connect } from 'react-redux'
+import { data as sd } from 'sharify'
+import colors from '@artsy/reaction-force/dist/Assets/Colors'
+import $ from 'jquery'
+
+const icons = () => require('./icons.jade')(...arguments)
+
+export class ArticleList extends Component {
+  static propTypes = {
+    display: PropTypes.string,
+    checkable: PropTypes.bool,
+    articles: PropTypes.array,
+    selected: PropTypes.func,
+    activeSessions: PropTypes.object,
+    user: PropTypes.object
+  }
+  getDisplayAttrs (article) {
+    if ((this.props.display === 'email') && article.email_metadata) {
+      return {
+        headline: article.email_metadata.headline,
+        image: article.email_metadata.image_url
+      }
+    } else {
+      return {
+        headline: article.thumbnail_title,
+        image: article.thumbnail_image
+      }
+    }
+  }
+
+  publishText (result) {
+    if (result.published_at && result.published) {
+      return `Published ${moment(result.published_at).fromNow()}`
+    } else if (result.scheduled_publish_at) {
+      return 'Scheduled to publish ' +
+        `${moment(result.scheduled_publish_at).fromNow()}`
+    } else {
+      return 'Last saved ' +
+        `${moment(result.updated_at).fromNow()}`
+    }
+  }
+
+  currentSessionText (article) {
+    const session = this.props.activeSessions[article.id]
+    return (
+      <h2>
+        <span>Last saved {moment(session.timestamp).fromNow()}</span>
+        <span className='name'><span className='circle' /> {session.user.name}</span>
+      </h2>
+    )
+  }
+
+  renderArticles () {
+    const { checkable, articles, activeSessions, user } = this.props
+    return articles.map(article => {
+      const attrs = this.getDisplayAttrs(article)
+      const session = activeSessions[article.id]
+      const isCurrentlyBeingEdited = session
+      const isCurrentUserEditing = user && session && user.id === session.user.id
+      const style = isCurrentlyBeingEdited ? {color: colors.grayMedium} : null
+      const shouldLockEditing = isCurrentlyBeingEdited && !isCurrentUserEditing
+      const lockedClass = shouldLockEditing ? 'locked' : ''
+
+      return (
+        <div style={style} className='article-list__result paginated-list-item' key={article.id}>
```

Comment:
We've been moving towards using StyledComponents rather than declaring inline styles.

Reply:
@eessex - thats great 👍 One thing I recently solved is better classname support during development -- i found that `babel-plugin-styled-components` displayName output isn't compatable with `babel-plugin-rewire`, but its easy enough to fix by making sure that babel-plugin-rewire is only loaded when `NODE_ENV=test`: https://github.com/artsy/force/blob/master/.babelrc. Checked positron's .babelrc and it seems like this still needs to be setup.

Reply:
Right, I didn't use styled-components here as this component was converted from coffeescript, I just focused on making it work in ES6 due to time. But I can definitely go back after we this merged in and convert the styling as well.

@damassi I wasn't aware of `babel-plugin-styled-components`, very useful for debugging purposes.

Reply:
Ok not a blocker for this PR but maybe add a TODO so we don't forget